### PR TITLE
[IMP] Add check Duplicate field in xml record

### PR DESCRIPTION
**Pull request by lescobarvx**
[IMP] Add check Duplicate field in xml record

### CODE REVIEW

File: pylint_odoo/test_repo/broken_module/model_view_odoo2.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <!-- duplicate record id -->
+        <record id="view_model_form" model="ir.ui.view">
+            <field name="name">view.model.form</field>
+            <field name="model">test.model</field>
+            <field name="arch" type="xml">
+                <tree string="Test model">
+                    <field name="name"/>
```
Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/model_view_odoo2.xml]: Chamo, no había considerado el caso de subvistas...
Es decir
```xml
<record....
   <field name="arch"...
       <form string="Test model">
           <field name="name"/>
           <field name="field_ids" widget="one2many_list">
                  <tree>
                       <field name="name"/>
                  </tree>
           </field>
```

ojo este es solo un ejemplo, busca uno real...
Aquí no debería causar un error... checa como lo podemos solucionar... (poniendo el tree name completo, para la comparación, maybe)

Comment (lescobarvx) [inline pylint_odoo/test_repo/broken_module/model_view_odoo2.xml]: @moylop260 
es decir, validar si la subvista tambien tiene fields duplicados??

Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/model_view_odoo2.xml]: Sí we

File: pylint_odoo/test_repo/broken_module/model_view_odoo2.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        </record>
+
+        <!-- 2 duplicate record field - "arch" in record and "model" in subview -->
+        <record id="view_model_form3" model="ir.ui.view">
+            <field name="arch">test.model5</field>
+            <field name="name">view.model.form</field>
+            <field name="model">test.model</field>
+            <field name="arch" type="xml">
+                <tree string="Test model6">
+                    <field name="model">test.model5</field>
```
Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/model_view_odoo2.xml]: Add tests case with subviews

**Comment by moylop260** [comment]
The CI show: 
https://travis-ci.org/Vauxoo/pylint-odoo/builds/123221289#L525
`pylint_odoo/test_repo/broken_module/__init__.py:1: [W7907(duplicate-xml-fields), ] Duplicate xml field "broken_module.view_model_form3.arch"`
But arch is not duplicated

File: pylint_odoo/test_repo/broken_module/model_view_odoo2.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                <tree string="Test model4">
+                    <field name="name">subview.model.form</field>
+                    <field model="model">test.model4</field>
+                </tree>
+            </field>
+        </record>
+
+        <!-- duplicate record field "arch" -->
+        <record id="view_model_form3" model="ir.ui.view">
+            <field name="arch">test.model5</field>
```
Comment (lescobarvx) [inline pylint_odoo/test_repo/broken_module/model_view_odoo2.xml]: @moylop260 
here is duplicated "arch" field

Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/model_view_odoo2.xml]: Thats is a good point
Thank you

**Comment by moylop260** [comment]
LGTM please, create pr to OCA

**Comment by lescobarvx** [comment]
@moylop260 
ready change where no checks inherited views

File: pylint_odoo/test_repo/broken_module/model_view_odoo2.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+        <!-- inherit view is not checked -->
+        <record id="view_model_form2" model="ir.ui.view">
+            <field name="name">view.model.form</field>
+            <field name="model">test.model</field>
+            <field name="inherit_id" ref="id_category_list"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='description']" position="after">
+                    <field name="description"/>
+                    <field name="user_id"/>
```
Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/model_view_odoo2.xml]: Add a inherit with duplicated and a inherit without duplicated

**Comment by lescobarvx** [comment]
@moylop260 
Ready, 2 test case for inherited views: with duplicate fields and without duplicate fields

**Comment by moylop260** [comment]
:+1:

**Comment by moylop260** [comment]
Could you create the dummies pr?

**Pull request merged**